pallets/liquid-staking/src/mock.rs

```diff
@@ -1,29 +1,225 @@
+    ChildParachainAsNative, ChildParachainConvertsVia, ChildSystemParachainAsSuperuser,
+    CurrencyAdapter as XcmCurrencyAdapter, EnsureXcmOrigin, FixedRateOfConcreteFungible,
+    FixedWeightBounds, IsConcrete, LocationInverter, NativeAsset, ParentAsSuperuser,
+    ParentIsDefault, RelayChainAsNative, SiblingParachainAsNative, SiblingParachainConvertsVia,
+    SignedAccountId32AsNative, SignedToAccountId32, SovereignSignedViaLocation, TakeWeightCredit,
+};
+use xcm_executor::{Config, XcmExecutor};
+use xcm_simulator::{decl_test_network, decl_test_parachain, decl_test_relay_chain};
+pub type AccountId = AccountId32;
+pub use westend_runtime;
+
+parameter_types! {
+    pub const ReservedXcmpWeight: Weight = WEIGHT_PER_SECOND / 4;
+    pub const ReservedDmpWeight: Weight = WEIGHT_PER_SECOND / 4;
+}
+
+impl cumulus_pallet_parachain_system::Config for Test {
+    type Event = Event;
+    type OnValidationData = ();
+    type SelfParaId = ParachainInfo;
+    type DmpMessageHandler = DmpQueue;
+    type ReservedDmpWeight = ReservedDmpWeight;
+    type OutboundXcmpMessageSource = XcmpQueue;
+    type XcmpMessageHandler = XcmpQueue;
+    type ReservedXcmpWeight = ReservedXcmpWeight;
+}
+
+impl parachain_info::Config for Test {}
+
+parameter_types! {
+    pub const DotLocation: MultiLocation = MultiLocation::X1(Parent);
+    pub const RelayNetwork: NetworkId = NetworkId::Kusama;
+    pub RelayChainOrigin: Origin = cumulus_pallet_xcm::Origin::Relay.into();
+    pub Ancestry: MultiLocation = Parachain(ParachainInfo::parachain_id().into()).into();
+}
+
+pub type LocationToAccountId = (
+    ParentIsDefault<AccountId>,
+    SiblingParachainConvertsVia<Sibling, AccountId>,
+    AccountId32Aliases<RelayNetwork, AccountId>,
+);
+
+pub type XcmOriginToCallOrigin = (
+    SovereignSignedViaLocation<LocationToAccountId, Origin>,
+    RelayChainAsNative<RelayChainOrigin, Origin>,
+    SiblingParachainAsNative<cumulus_pallet_xcm::Origin, Origin>,
+    SignedAccountId32AsNative<RelayNetwork, Origin>,
+    XcmPassthrough<Origin>,
+);
+
+parameter_types! {
+    pub const UnitWeightCost: Weight = 1;
+    pub DotPerSecond: (MultiLocation, u128) = (X1(Parent), 1);
+}
+
+// pub type LocalAssetTransactor =
+//     XcmCurrencyAdapter<ParaTokens::CurrencyAdapter, IsConcrete<DotLocation>, LocationToAccountId, AccountId, ()>;
+
+pub type LocalAssetTransactor = MultiCurrencyAdapter<
+    Tokens,
+    (),
+    IsNativeConcrete<CurrencyId, CurrencyIdConvert>,
+    AccountId,
+    LocationToAccountId,
+    CurrencyId,
+    CurrencyIdConvert,
+>;
+
+pub type XcmRouter = ParachainXcmRouter<ParachainInfo>;
+pub type Barrier = AllowUnpaidExecutionFrom<Everything>;
+
+pub struct XcmConfig;
+impl Config for XcmConfig {
+    type Call = Call;
+    type XcmSender = XcmRouter;
+    type AssetTransactor = LocalAssetTransactor;
+    type OriginConverter = XcmOriginToCallOrigin;
+    type IsReserve = NativeAsset;
```

Comment:
I think here we need `MultiNativeAsset`

Reply:
`NativeAsset` is good for now, because when test staking methods, `KSM` is the only asset that we need to consider.